test/extended/util/disruption/backend_sampler_tester.go

```diff
@@ -59,14 +60,23 @@ func (t *backendDisruptionTest) WithPostTeardown(postTearDown TearDownFunc) *bac
 	return t
 }
 
-func (t *backendDisruptionTest) historicalP95Disruption(f *framework.Framework, totalDuration time.Duration) (*time.Duration, string, error) {
+func (t *backendDisruptionTest) historicalP99Disruption(f *framework.Framework, totalDuration time.Duration) (*time.Duration, string, error) {
 	backendName := t.backend.GetDisruptionBackendName() + "-" + string(t.backend.GetConnectionType()) + "-connections"
 	jobType, err := platformidentification.GetJobType(context.TODO(), f.ClientConfig())
 	if err != nil {
 		return nil, "", err
 	}
 
-	return allowedbackenddisruption.GetAllowedDisruption(backendName, *jobType)
+	extraMatchers := []historicaldata.NextBestKey{}
+	if jobType.Topology == "single" {
+		extraMatchers = append(
+			extraMatchers,
+			historicaldata.OnSDN,
+			historicaldata.Combine(historicaldata.PreviousReleaseUpgrade, historicaldata.OnSDN),
+		)
+	}
```

Comment:
It looks like this should be an OnTopology matcher similar to OnArchitecture (which has access to JobType), no public Combine, no extraMatchers on all function arguments. Would this work?

Reply:
That's a good idea, I reworked this and took your suggestion in. Slight change I made was inverting the logic and wording of that `OnTopology` intention, since we want to match on `sdn` but only for single nodes. What do you think of the change?

Reply:
Looks much better to me, we'll need David to take a look when he's back from PTO now.

Reply:
That sounds great, thanks!